lib/boot-without-auth.tsx

```diff
@@ -40,6 +40,8 @@ class AppWithoutAuth extends Component<Props, State> {
       return;
     }
 
+    username = username.trim().toLowerCase();
```

Comment:
this is probably going to sound really nit-picky but shouldn't have any cases any more of this kind of mutation. would you mind splitting this into an _immutable_ change so we don't have "the value of `username` above this arbitrary line" and also "the value of `username` below this arbitrary line?"

```ts
authenticate = (givenUsername: string, password: string) => {
	…
	const username = givenUsername.trim().toLowerCase();
```

…or skip the update and only transform the username where it matters, during validation…

```ts
.authorize(username.trim().toLowerCase(), password)
.then(…this.props.onAuth(user.access_token, username, false))
```